### ISSUE TITLE
Remove "extension" from AJAX results

### ISSUE THREAD

**Issue by dereks** [servocoder/RichFilemanager#106]
Currently the JSON returned by the server contains an `attributes` member called `extension` for every file entry:
```json
"extension":"txt",
```
This is a calculated, redundant value that can be extracted from `attributes.name`. The Javascript already calculates extensions with a function called `getExtension()` for other purposes.

To normalize the response data, guarantee consistency, reduce code size, and reduce response size, we could eliminate this member. There are nine references to `resourceObject.attributes.extension` that would be replaced in the JS. For example:

```js
if(getExtension(file.name) != resourceObject.attributes.extension) {
	fm.error(lg.ERROR_REPLACING_FILE + " ." + resourceObject.attributes.extension);
	return false;
}
```
...becomes...
```js
if(getExtension(file.name) != getExtension(resourceObject.attributes.name)) {
	fm.error(lg.ERROR_REPLACING_FILE + " ." + getExtension(resourceObject.attributes.name));
	return false;
}
```

This is a low-priority issue that I would consider minor code cleanup. I'd be happy to submit a Pull Request if you approve of this change. Thanks!

**Comment by servocoder** [comment]
Thanks for the suggestion.
Ok, but let's postpone this modification for a bit.
I don't want to change any things related to the connector at the moment. A dependent code is going to be committed soon.

**Comment by servocoder** [comment]
Released in v2.4.0
